src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -85,5 +133,218 @@ public override bool Execute()
 
             return true;
         }
+
+        private void AddDefaultMappingsForOnlineSources(XElement pkgSrcMappingClearElement, XElement pkgSourcesElement)
+        {
+            foreach (string sourceName in pkgSourcesElement
+                .Descendants()
+                .Where(e => e.Name == "add" && !allSourcesPackages.Keys.Contains(e.Attribute("key").Value))
+                .Select(e => e.Attribute("key").Value)
+                .Distinct())
+            {
+                pkgSrcMappingClearElement.AddAfterSelf(new XElement("packageSource", new XAttribute("key", sourceName), new XElement("package", new XAttribute("pattern", "*"))));
+            }
+        }
+
+        private XElement GetPackageMappingsElementForSource(string key, List<string> value)
+        {
+            XElement pkgSrc = new XElement("packageSource", new XAttribute("key", key));
+            foreach (string pattern in value)
+            {
+                pkgSrc.Add(new XElement("package", new XAttribute("pattern", pattern)));
+            }
+
+            return pkgSrc;
+        }
+
+        private XElement GetPackageMappingsElementForSource(string packageSource)
+        {
+            bool isCurrentSourceBuiltSource =
+                packageSource.StartsWith("source-built-") ||
```

Comment:
The `source-built-` string should be passed as a task property so it can be shared with the value from here:

https://github.com/dotnet/installer/blob/79f35dcddc1a4c48fdd46e55b9563c5d9f4cba6e/src/SourceBuild/content/repo-projects/Directory.Build.targets#L81